lib/hunter/commands/send.rb

```diff
@@ -66,25 +66,36 @@ def run
 
           request.body = data.to_json
 
-          begin
-            response = http.request(request)
-            response.value
-            puts "Successful transmission"
-          rescue Errno::ECONNREFUSED => e
-            puts "The server is unavailable"
-            puts e.message
-          rescue Net::HTTPServerException => e
-            if response.code == "401"
-              raise "Authentication key mismatch"
-            else
-              raise "Unknown HTTP error"
-            end
-          end
+          send_request(http, request)
         end
       end
 
       private
 
+      def send_request(http, request)
+        begin
+          response = http.request(request)
+          response.value
+          puts "Successful transmission"
+          return
+        rescue Errno::ECONNREFUSED => e
+          msg = "The server is unavailable\n" + e.message
+        rescue Net::HTTPServerException => e
+          if response.code == "401"
+            msg = "Authentication key mismatch"
+          else
+            msg = "Unknown HTTP error"
+          end
+        end
+        if @options.retry
+          puts msg
+          sleep(@options.retry)
+          send_request(http, request)
```

Comment:
This produces a rather dangerous situation. We're recursively calling `send_request` with no base case. If the host that the system is looking for never returns a success status, we will be calling `send_request` recursively, forever, until this process holds all of the system resources.

We should change this recursive method to an iterative method. I don't mind it looping ad infinitum; it's down to the user to monitor their system services.